client/blocks/payment-request/hooks.js

```diff
@@ -82,15 +101,19 @@ export const useShippingAddressUpdateHandler = (
 	paymentRequestType
 ) => {
 	useEffect( () => {
-		// Need to use `?.` here in case paymentRequest is null.
-		const shippingAddressUpdateHandler = paymentRequest?.on(
-			'shippingaddresschange',
-			shippingAddressChangeHandler( paymentRequestType )
-		);
+		let handler = noop;
+
+		if ( paymentRequest ) {
+			handler = paymentRequest.on(
+				'shippingaddresschange',
+				shippingAddressChangeHandler( paymentRequestType )
+			);
+		}
 
 		return () => {
-			// Need to use `?.` here in case shippingAddressUpdateHandler is null.
-			shippingAddressUpdateHandler?.removeAllListeners();
+			if ( paymentRequest ) {
```

Comment:
The `noop` is an alternative to the optional chaining and ensures the `removeEventListener` always exists, even if these conditions change in the future, or if the unmount function runs for whatever reason before the handler is set. The noop in the original blocks was for `removeAllListeners`, but we don't use that here.

Reply:
The cleanup function isn't created until after the handler is assigned, but if we'd like to allow for the condition to change in the future, that sounds fine to me :+1:

In that case, I don't see the need for this 2nd `if` statement, since it's already been ensured that the `removeEventListener` will exist, as you point out. Not a big deal, since it's harmless to include it, but it would be nice to keep the code a bit tighter (unless I'm missing something about why we need it)

Reply:
Kept the `noop` and removed the second `if` statement in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1603/commits/ed579b6d5d463e13b5806986990a1e2bd99c6819